app/webhook.go

```diff
@@ -789,10 +789,29 @@ func (a *App) HandleIncomingWebhook(c *request.Context, hookID string, req *mode
 		overrideIconURL = req.IconURL
 	}
 
-	_, err := a.CreateWebhookPost(c, hook.UserId, channel, text, overrideUsername, overrideIconURL, req.IconEmoji, req.Props, webhookType, "")
+	postingUserID := a.GetBotUserId(hook.UserId, req.Username)
+	if postingUserID == nil {
+		postingUserID = &hook.UserId
+	}
+
+	_, err := a.CreateWebhookPost(c, *postingUserID, channel, text, overrideUsername, overrideIconURL, req.IconEmoji, req.Props, webhookType, "")
 	return err
 }
 
+func (a *App) GetBotUserId(userID string, botUsername string) *string {
+	bots, err := a.Srv().Store().Bot().GetAll(&model.BotGetOptions{Page: 0, PerPage: 10})
+	if err != nil {
+		return nil
+	}
+	for i := range bots {
+		bot := bots[i]
+		if bot.OwnerId == userID && bot.Username == botUsername {
+			return &bot.UserId
+		}
+	}
+	return nil
+}
```

Comment:
We can have this have the signature `(string, error)`, and return `"", nil` in the "no bot id available" case. The error from `GetAll` should be logged either by this function or its caller.